### ISSUE TITLE
Bot no longer receives events after reconnect

### ISSUE THREAD

**Issue by valzargaming** [teamreflex/DiscordPHP#454]
**Environment**
- PHP Version:
    - 7.4.3
- DiscordPHP Version:
    - Lately

**Describe the bug**
I used my bot's ;serverstatus command to query 2 different servers, one of those servers being local and targeting both an open and closed port with the following CURL settings:
```php
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1); //return the transfer as a string
curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 3);
curl_setopt($ch, CURLOPT_TIMEOUT, 10);
```
After receiving data I then try to ->sendEmbed on a channel, but instead of the messaage being sent I am greeted with a websocket closed which tries to reconnect.
![image](https://user-images.githubusercontent.com/7202504/101615688-6a10b400-39dc-11eb-9125-819c5b49df9d.png)
After this point, the bot stops emitting events from Discord until I manually restart it.

**Comment by valzargaming** [comment]
I attached an HTTP logger and ran the same function again, this time without sending any messages after serverstatus command until the CURL event finishes. The bot acknowledges that the message event is received, and sees the ping command, but its attempt to respond to it fails. After this point is when the bot stops emitting the message event at all.
![image](https://user-images.githubusercontent.com/7202504/101616633-9c6ee100-39dd-11eb-84dd-93ff431039ca.png)

**Comment by valzargaming** [comment]
For whatever it's worth, I changed the majority of my code to utilize only the code snippet below, skipping the image generation completely.
```php
$server = socket_create(AF_INET,SOCK_STREAM,SOL_TCP);
socket_set_option($server, SOL_SOCKET, SO_RCVTIMEO, array('sec'=>2, 'usec'=>0));
socket_set_option($server, SOL_SOCKET, SO_SNDTIMEO, array('sec'=>2, 'usec'=>0)); //Timeout, may need tweaks
```
The main differences is that I was using curl in addition to this to send a GET request to my website which generates an image using the same socket code, which is what I linked to in my embed. Generating the image takes some time, so I think using the web URL http://www.valzargaming.com/servers/gamebanner.php?servernum=0 may have been causing it to wait while it retrieves the image, I'm not really sure.
